File: Euler7.js
//euler7
let primes = function(n){
	let r=[]
	let a=Array.from(new Array(n+1),x=>true)
	a[0]=false
	a[1]=false
	for(let i=2;i<=n;i++){
		if(a[i]){
			r.push(i)
			for(let j=2;j*i<=n;j++){
				a[j*i]=false
			}
		}
	}
	return r
}

console.log(primes(1000000)[10001-1])

File: Euler7.hs
isPrime :: Int -> Bool
isPrime 2 = True
isPrime n = and [n `mod` x /= 0|x<- [2]++[3,5..ceiling (sqrt (fromIntegral n))]]

euler7 :: Int -> Int
euler7 n = last (take n (filter isPrime [2..]))

main :: IO()
main = do
--  print (isPrime 7)
  print(euler7 10001)
